File: internal/domain/model/task.go
package model

import (
	"time"
)

type TaskStatus string

const (
	StatusActive    TaskStatus = "ACTIVE"
	StatusCompleted TaskStatus = "COMPLETED"
	StatusOverdue   TaskStatus = "OVERDUE"
	StatusLate      TaskStatus = "LATE"
)

type TaskPriority string

const (
	PriorityLow      TaskPriority = "LOW"
	PriorityMedium   TaskPriority = "MEDIUM"
	PriorityHigh     TaskPriority = "HIGH"
	PriorityCritical TaskPriority = "CRITICAL"
)

type TaskFilter struct {
	Status    string
	Priority  string
	SortBy    string
	SortOrder string
	Page      int
	PageSize  int
}

type Task struct {
	ID          string       `json:"id"`
	Title       string       `json:"title"`
	Description *string      `json:"description"`
	Deadline    *time.Time   `json:"deadline"`
	Status      TaskStatus   `json:"status"`
	Priority    TaskPriority `json:"priority"`
	CreatedAt   time.Time    `json:"created_at"`
	UpdatedAt   *time.Time   `json:"updated_at"`
	IsCompleted bool         `json:"is_completed"`
}

File: mobile/mobile/Features/Tasks/Models/Task.swift
import Foundation

struct Task: Identifiable, Codable {
    let id: String
    var title: String
    var description: String?
    var deadline: String?
    var status: TaskStatus
    var priority: TaskPriority
    let createdAt: String
    var updatedAt: String?
    
    enum CodingKeys: String, CodingKey {
        case id
        case title
        case description
        case deadline
        case status
        case priority
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}
